R/build_html_mans.R

```diff
@@ -0,0 +1,187 @@
+{
+    files <- Sys.glob(file.path(web_dir, "*", "aliases.rds"))
+    packages <- basename(dirname(files))
+    if (force || !is_file(aliases_db_file)) {
+        db <- lapply(files, readRDS)
+        names(db) <- packages
+    } else {
+        db <- readRDS(aliases_db_file)
+        ## Drop entries in db not in package web area.
+        db <- db[!is.na(match(names(db), packages))]
+        ## Update entries for which aliases file is more recent than the
+        ## db file.
+        mtimes <- file.mtime(files)
+        files <- files[mtimes >= file.mtime(aliases_db_file)]
+        db[basename(dirname(files))] <- lapply(files, readRDS)
+    }
+
+    db[sort(names(db))]
+}
+
+is_file <- function(x) file.exists(x) && !file.info(x)[["isdir"]]
+
+#' @rdname html-manuals
+#'
+#' @param rdxrefs_db_file `character(1)` The file path to `rdxrefs.rds` file
+#'   generated by the `build_db_from_source` function.
+#'
+#' @examples
+#' if (interactive()) {
+#'     src_base <- "~/minibioc/packages/3.20/bioc/"
+#'     web_dir <- file.path(src_base, "web", "packages")
+#'
+#'     meta_folder <- file.path(contrib.url(src_base), "Meta")
+#'     if (!dir.exists(meta_folder)) dir.create(meta_folder, recursive = TRUE)
+#'     rdxrefs_db_file <- file.path(meta_folder, "rdxrefs.rds")
+#'
+#'     meta_rdxrefs_db <- build_meta_rdxrefs_db(web_dir, rdxrefs_db_file)
+#'
+#'     saveRDS(meta_rdxrefs_db, rdxrefs_db_file, version = 2)
+#' }
+#' @export
+build_meta_rdxrefs_db <-
+    function(web_dir, rdxrefs_db_file, force = FALSE)
+{
+    files <- Sys.glob(file.path(web_dir, "*", "rdxrefs.rds"))
+    packages <- basename(dirname(files))
+    if(force || !is_file(rdxrefs_db_file)) {
+        db <- lapply(files, readRDS)
+        names(db) <- packages
+    } else {
+        db <- readRDS(rdxrefs_db_file)
+        ## Drop entries in db not in package web area.
+        db <- db[!is.na(match(names(db), packages))]
+        ## Update entries for which rdxrefs file is more recent than the
+        ## db file.
+        mtimes <- file.mtime(files)
+        files <- files[mtimes >= file.mtime(rdxrefs_db_file)]
+        db[basename(dirname(files))] <- lapply(files, readRDS)
+    }
+
+    db[sort(names(db))]
+}
+
+#' @rdname html-manuals
+#'
+#' @param package_dirs `character()` A vector of local package directories.
+#'
+#' @examples
+#' if (interative()) {
+#'     library(BiocPkgTools)
+#'     bioc_sub <- pkgBiocDeps(
+#'         "SummarizedExperiment", pkgType = "software",
+#'         recursive = TRUE, only.bioc = TRUE
+#'     )
+#'     bioc_sub <- unlist(bioc_sub, use.names = FALSE)
+#'
+#'     ## generate from Bioc package source dirs
+#'     packages <- file.path(normalizePath("~/bioc"), bioc_sub)
+#'     src_base <- "~/minibioc/packages/3.20/bioc"
+#'
+#'     build_html_mans(packages, src_base)
+#' }
+#' @export
+build_html_mans <- function(package_dirs, src_base) {
+    packages <- basename(package_dirs)
+    html_dir <- file.path(src_base, "manuals")
```

Comment:
Could there be a little more flexibility in the path? There's already a `manuals` directory that's created on the BBS, which I may rewrite to "manuals/pdf" then I could do "manuals/html" for the html manuals?

Reply:
Why not put them both in the same folder? 

https://bioconductor.org/packages/release/bioc/manuals/MultiAssayExperiment/man/MultiAssayExperiment.pdf
I changed the code to create it under `manuals/pkg/man/pkg.html`.